hip/solver/batch_cg_kernels.hip.cpp

```diff
@@ -91,17 +110,44 @@ class KernelCaller {
     {
         using real_type = gko::remove_complex<value_type>;
         const size_type nbatch = a.num_batch;
-
-        const int shared_size =
-            gko::kernels::batch_cg::local_memory_requirement<value_type>(
-                a.num_rows, b.num_rhs) +
-            PrecType::dynamic_work_size(a.num_rows, a.num_nnz) *
-                sizeof(value_type);
-
+        const int shared_gap = ((a.num_rows - 1) / 8 + 1) * 8;
+        static_assert(default_block_size >= 2 * config::warp_size,
+                      "Need at least two warps!");
+
+        const int block_size =
+            get_num_threads_per_block<BatchMatrixType>(exec_, a.num_rows);
+        assert(block_size >= 2 * config::warp_size);
+        const size_t prec_size =
+            PrecType::dynamic_work_size(shared_gap, a.num_nnz) *
+            sizeof(value_type);
+        const int shmem_per_blk = exec_->get_max_shared_memory_per_block();
+        // std::cerr << " Num shmem per block: " << shmem_per_blk << "\n";
+        const auto sconf =
+            gko::kernels::batch_cg::compute_shared_storage<PrecType,
+                                                           value_type>(
+                shmem_per_blk, shared_gap, a.num_nnz, b.num_rhs);
+        const size_t shared_size =
+            sconf.n_shared * shared_gap * sizeof(value_type) +
+            (sconf.prec_shared ? prec_size : 0);
+        auto workspace = gko::Array<value_type>(
+            exec_, sconf.gmem_stride_bytes * nbatch / sizeof(value_type));
+        assert(sconf.gmem_stride_bytes % sizeof(value_type) == 0);
+
+        // std::cerr << " CG: vectors in shared memory = " << sconf.n_shared
+        //           << " gmem stride bytes " << sconf.gmem_stride_bytes <<
+        //           "\n";
+        // if (sconf.prec_shared) {
+        //     std::cerr << " CG: precondiioner is in shared memory.\n";
+        // }
+        // std::cerr << " CG: vectors in global memory = " << sconf.n_global
+        //           << "\n Hip: number of threads per warp = "
+        //           << config::warp_size
+        //           << "\n CG: number of threads per block = " << block_size
+        //           << "\n";
```

Comment:
Is debugging over? Alternatively, that can be moved to a function if you think you might reuse this later

Reply:
Yeah, I think I would keep it for now and remove it once we plan to merge into develop.